### PR TITLE
Fixes related to PASE-only commissioning on Android

### PR DESCRIPTION
**Pull request by chrisdecenzo**
#### Problem
* PASE-only commissioning is not supported by AutoCommissioner
* Android commissioning calls (commissionDevice, pairDevice, pairDeviceWithAddress) are resetting the commissioner conflguration

#### Change overview
* add configuration for PASE-only commissioning
* make Android commissioning calls use the existing commissioner configuration (rather than reset to a partial configuration)

#### Testing
* Tested using Android commissioning

### CODE REVIEW

File: src/controller/AutoCommissioner.cpp [hunk 1]
```diff
@@ -280,14 +290,22 @@ CommissioningStage AutoCommissioner::GetNextCommissioningStageInternal(Commissio
...
             return CommissioningStage::kFindOperational;
         }
     case CommissioningStage::kThreadNetworkEnable:
+        if (mParams.GetPASEOnlyCommissioning().ValueOr(false))
+        {
+            return CommissioningStage::kCleanup;
+        }
         return CommissioningStage::kFindOperational;
     case CommissioningStage::kFindOperational:
-        return CommissioningStage::kSendComplete;
```
Comment (chrisdecenzo) [inline src/controller/AutoCommissioner.cpp:290]: this is a bug

**Review by sharadb-amazon — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Review by bzbarsky-apple — DISMISSED** [review]
The naming of the new member and APIs is a bit confusing.  Since they're all about skipping sending CommissioningComplete, naming them accordingly would be good.

File: src/controller/CHIPDeviceController.cpp [hunk 2]
```diff
@@ -1551,6 +1551,11 @@ void DeviceCommissioner::CommissioningStageComplete(CHIP_ERROR err, Commissionin
 {
     // Once this stage is complete, reset mDeviceBeingCommissioned - this will be reset when the delegate calls the next step.
     MATTER_TRACE_EVENT_SCOPE("CommissioningStageComplete", "DeviceCommissioner");
+    if (mDeviceBeingCommissioned == nullptr)
+    {
+        ChipLogProgress(Controller, "CommissioningStageComplete with null mDeviceBeingCommissioned");
```
Comment (bzbarsky-apple) [inline src/controller/CHIPDeviceController.cpp:1556]: Under what conditions was this being hit?

File: src/controller/CommissioningDelegate.h [hunk 3]
```diff
@@ -407,6 +416,7 @@ class CommissioningParameters
         nullptr; // Delegate to handle device attestation failures during commissioning
     Optional<bool> mAttemptWiFiNetworkScan;
     Optional<bool> mAttemptThreadNetworkScan; // This automatically gets set to false when a ThreadOperationalDataset is set
+    Optional<bool> mPASEOnlyCommissioning;
```
Comment (bzbarsky-apple) [inline src/controller/CommissioningDelegate.h:419]: I think this would be clearer as mSkipCommissioningComplete or so.

**Review by hawk248 — APPROVED** [review]

File: src/controller/CommissioningDelegate.h (hunk 3 above)
Comment (chrisdecenzo) [reply src/controller/CommissioningDelegate.h]: updated name

File: src/controller/CHIPDeviceController.cpp (hunk 2 above)
Comment (chrisdecenzo) [reply src/controller/CHIPDeviceController.cpp:1566]: when cleaning up and operational discovery is skipped, CommissioningStageComplete gets called twice.

**Pull request merged**